app/command.go

```diff
@@ -176,24 +176,45 @@ func (a *App) ExecuteCommand(args *model.CommandArgs) (*model.CommandResponse, *
 
 	args.TriggerId = triggerId
 
-	if provider != nil {
-		if cmd := provider.GetCommand(a, args.T); cmd != nil {
-			response := provider.DoCommand(a, args, message)
-			return a.HandleCommandResponse(cmd, args, response, true)
-		}
+	if cmd, response := a.tryExecuteBuiltInCommand(args, trigger, message); cmd != nil && response != nil {
+		return a.HandleCommandResponse(cmd, args, response, true)
 	}
 
-	cmd, response, appErr := a.ExecutePluginCommand(args)
-	if appErr != nil {
+	if cmd, response, appErr := a.ExecutePluginCommand(args); appErr != nil {
```

Comment:
I am not sure if we have a preferred style recommendation for "horizontal" vs "vertical" error handling. I have a personal preference for the latter, also consistent with https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow, but most important I think we ought to be consistent.

Reply:
I can change that. I forgot we had a recent push to change it to be that way, so I think I've actually reverted some recent work to do that. Do you have any preference on whether the if two statements would be separate as before or if they would use an else if? I kind of like the logical grouping provided by else if